client/ayon_core/hosts/houdini/plugins/publish/extract_mantra_ifd.py

```diff
@@ -22,20 +22,13 @@ def process(self, instance):
 
         ropnode = hou.node(instance.data.get("instance_node"))
         output = ropnode.evalParm("soho_diskfile")
-        staging_dir = os.path.dirname(output)
+        staging_dir, file_name = os.path.split(output)
         instance.data["stagingDir"] = staging_dir
+        # render rop
+        self.log.debug("Writing IFD '{}' to '{}'".format(file_name, staging_dir))
+        render_rop(ropnode)
 
         files = instance.data["frames"]
-        missing_frames = [
-            frame
-            for frame in instance.data["frames"]
-            if not os.path.exists(
-                os.path.normpath(os.path.join(staging_dir, frame)))
-        ]
-        if missing_frames:
-            raise RuntimeError("Failed to complete Mantra ifd extraction. "
-                               "Missing output files: {}".format(
-                                   missing_frames))
```

Comment:
Note that a `render_rop` can't always accurately detect whether it failed or whether a user cancelled. A user cancelling the render _may_ just exit the function without an error making it look like it finished. This "check" logic here was once intended to cover some very simple 'check cases' that if the files weren't there we were pretty sure the full render did not succeed.

I think it happened on certain ROPs when cancelling super early or JUST as it went from rendering one frame to another. But still, you wouldn't want the publisher to think it succeeded.

I had reported it to SideFX at the time and I haven't seen any report as of yet that they fixed that.

Reply:
Thanks for informing me! 
I've also furtherly checked it and found 2 things. 
1. If a user interrupts a render, this error will be raised  https://github.com/ynput/ayon-core/blob/7b2dc2676f04cd845669dcad9007b1f1e115728e/client/ayon_core/hosts/houdini/api/lib.py#L253
2. I add a try before `render_rop(ropnode)` in `extract_mantra_ifd` in order to see missing frames logic runs. it works.


Both errors are `RuntimeError` and they don't look good in the UI. 
I tried raising `PublishValidationError` instead but it doesn't work in the same manner as in validations. 

A question may be for @iLLiCiTiT : Do we have a generic `PublishError` that can be used in the same fashion as validation errors ? 

![image](https://github.com/ynput/ayon-core/assets/20871534/bfea5b0b-e22f-4dd6-b2b0-734e17add991)

Reply:
Ok, I did some experiment here.
It's outside the scope of this PR but this PR was a good example to do the experiment.
This one uses `KnownPublishError` 
![image](https://github.com/ynput/ayon-core/assets/20871534/dc1ebb2c-8a48-4225-9f97-647585d1e992)

tbh, I was overwhelmed I expected to have different `report_label`
e.g. 
> I did some ugly modifications in order to have this screenshot. by setting the `ReportsWidget.CrashWidget.report_label` with the `publish_error_msg` 
![image](https://github.com/ynput/ayon-core/assets/20871534/19eb01f1-4f90-4a18-9acc-846f5995a51e)

But, Truly I expected some general `PublishError` like `PublishValidationError` I did one modification to create the following screenshot (I removed that line and then used `PublishValidationError` as usual) . 

![image](https://github.com/ynput/ayon-core/assets/20871534/2775c4cd-8e73-4a9a-8032-4837acc0c908)

Reply:
I'm going to create a GH issue.. It would be a better place for such a discussion.

Reply:
Well - it's been requested quite frequently to improve the error view for anything that's not a validation report message. It's "too dumbed down" usually. Also a point @fabiaserra I believe has brought up quite a bit - and some others on discord as well. So thanks for tracking this as an issue. Would be good to include Dee Coureau into the discussion as well.